libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionColumn.php

```diff
@@ -27,11 +32,21 @@ public function getBaseType(): ?string
         return $this->baseType;
     }
 
+    public function getDataTypeDefinition(): ?DefinitionInterface
+    {
+        return $this->dataTypeDefinition;
+    }
+
     public function toArray()
     {
-        return [
+        $output = [
             'name' => $this->name,
-            'basetype' => $this->baseType,
         ];
+        if ($this->dataTypeDefinition) {
```

Comment:
well I understand what it does, but I don't really like the structure at all - there is a nullable class name or class being bubbled through 10 methods and 4 classes just to reach an if which magically makes the class return the right thing. I can't get rid of the feeling that the abstractions are not doing their job properly, maybe I'm to hungry before dinner.

Reply:
yeah, maybe @MiroCillik has some idea of how we can improve it.

Reply:
Maybe have BaseTypeTableDefinition and NativeTypeTableDefinition or sth like that

Reply:
updated